datasets/opus_sardware/opus_sardware.py

```diff
@@ -0,0 +1,98 @@
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Sardware - English to Sardinian"""
+
+from __future__ import absolute_import, division, print_function
+
+import os
+
+import datasets
+
+
+_CITATION = """\
+J. Tiedemann, 2012, Parallel Data, Tools and Interfaces in OPUS. \
+In Proceedings of the 8th International Conference on Language Resources and Evaluation (LREC 2012)
+"""
+
+
+_DESCRIPTION = """\
+This dataset is designed for machine translation from English to Sardinian."""
+
+
+_HOMEPAGE = "http://opus.nlpl.eu/sardware-v1.php"
+
+
+_LICENSE = ""
+
+
+_URLs = {
+    "train": "https://object.pouta.csc.fi/OPUS-sardware/v1/moses/en-sc.txt.zip",
+}
+
+
+class OpusSardware(datasets.GeneratorBasedBuilder):
+    """Sardware - English to Sardinian"""
+
+    VERSION = datasets.Version("1.0.0")
+
+    BUILDER_CONFIGS = [
+        datasets.BuilderConfig(name="en-sc", version=VERSION, description="Sardware - English to Sardinian")
+    ]
+
+    def _info(self):
+        return datasets.DatasetInfo(
+            description=_DESCRIPTION,
+            features=datasets.Features(
+                {"translation": datasets.features.Translation(languages=tuple(self.config.name.split("-")))}
+            ),
+            supervised_keys=None,
+            homepage="http://opus.nlpl.eu/sardware-v1.php",
+            citation=_CITATION,
+        )
+
+    def _split_generators(self, dl_manager):
+        """Returns SplitGenerators."""
+        data_dir = dl_manager.download_and_extract(_URLs)
+        return [
+            datasets.SplitGenerator(
+                name=datasets.Split.TRAIN,
+                # These kwargs will be passed to _generate_examples
+                gen_kwargs={
+                    "source_file": os.path.join(data_dir["train"], "sardware.en-sc.en"),
+                    "target_file": os.path.join(data_dir["train"], "sardware.en-sc.sc"),
+                    "split": "train",
+                },
+            ),
+        ]
+
+    def _generate_examples(self, source_file, target_file, split):
+        """This function returns the examples in the raw (text) form."""
+        with open(source_file, encoding="utf-8") as f:
+            source_sentences = f.read().split("\n")
+        with open(target_file, encoding="utf-8") as f:
+            target_sentences = f.read().split("\n")
+
+        assert len(target_sentences) == len(source_sentences), "Sizes do not match: %d vs %d for %s vs %s." % (
+            len(source_sentences),
+            len(target_sentences),
+            source_file,
+            target_file,
+        )
+
+        source, target = tuple(self.config.name.split("-"))
+        for idx, (l1, l2) in enumerate(zip(source_sentences, target_sentences)):
+            result = {"translation": {source: l1, target: l2}}
+            if all(result.values()):
+                yield idx, result
```

Comment:
```suggestion
            yield idx, result
```
the if statement is always true here

Reply:
let me know if you had something in mind for this, otherwise we can commit this change and merge